applications/StructuralMechanicsApplication/custom_utilities/structural_mechanics_element_utilities.cpp

```diff
@@ -116,6 +116,83 @@ void CalculateRayleighDampingMatrix(
     KRATOS_CATCH( "CalculateRayleighDampingMatrix" )
 }
 
+
+double CalculateCurrentLength2D2N(const Element& rElement) {
+  KRATOS_TRY;
+  const double numerical_limit = std::numeric_limits<double>::epsilon();
+  const double du =
+      rElement.GetGeometry()[1].FastGetSolutionStepValue(DISPLACEMENT_X) -
+      rElement.GetGeometry()[0].FastGetSolutionStepValue(DISPLACEMENT_X);
+  const double dv =
+      rElement.GetGeometry()[1].FastGetSolutionStepValue(DISPLACEMENT_Y) -
+      rElement.GetGeometry()[0].FastGetSolutionStepValue(DISPLACEMENT_Y);
+
+  const double dx = rElement.GetGeometry()[1].X0() - rElement.GetGeometry()[0].X0();
+  const double dy = rElement.GetGeometry()[1].Y0() - rElement.GetGeometry()[0].Y0();
+
+  const double l = std::sqrt((du + dx) * (du + dx) + (dv + dy) * (dv + dy));
+
+  KRATOS_ERROR_IF(l < numerical_limit) << "length 0 for element " << rElement.Id()
+                                       << std::endl;
+  return l;
+  KRATOS_CATCH("")
+}
+
+double CalculateReferenceLength2D2N(const Element& rElement) {
+  KRATOS_TRY;
+  const double numerical_limit = std::numeric_limits<double>::epsilon();
+  const double dx = rElement.GetGeometry()[1].X0() - rElement.GetGeometry()[0].X0();
+  const double dy = rElement.GetGeometry()[1].Y0() - rElement.GetGeometry()[0].Y0();
+  const double L = std::sqrt((dx * dx) + (dy * dy));
+
+  KRATOS_ERROR_IF(L < numerical_limit) << "length 0 for element " << rElement.Id()
+                                       << std::endl;
+  return L;
+  KRATOS_CATCH("")
+}
+
+
+double CalculateReferenceLength3D2N(const Element& rElement) {
+
+  KRATOS_TRY;
+  const double numerical_limit = std::numeric_limits<double>::epsilon();
+  const double dx = rElement.GetGeometry()[1].X0() - rElement.GetGeometry()[0].X0();
+  const double dy = rElement.GetGeometry()[1].Y0() - rElement.GetGeometry()[0].Y0();
+  const double dz = rElement.GetGeometry()[1].Z0() - rElement.GetGeometry()[0].Z0();
+  const double L = std::sqrt(dx * dx + dy * dy + dz * dz);
+
+  KRATOS_ERROR_IF(L<=numerical_limit)
+   << "Reference Length of element" << rElement.Id() << "~ 0" << std::endl;
+  return L;
+  KRATOS_CATCH("")
+}
+
+double CalculateCurrentLength3D2N(const Element& rElement) {
```

Comment:
Is this the Lenght() method of the geometry?

Reply:
nope it is the initial length + the displacements

We tried already using the methods from Geometry

Reply:
The ReferenceLength... is the length of the geometry, we could directly call that here

Reply:
Thanks